spec/src/main/resources/META-INF/smithy/jsonrpc/jsonrpc.smithy

```diff
@@ -0,0 +1,43 @@
+$version: "2"
+
+namespace jsonrpc
```

Comment:
this files defines the various traits that give meaning to the BSP specification in the context of jsonRPC.

Reply:
> how does that work if they want to make an official one 

There is no global trait index. In the same way that there is no official Json library in Scala. It might be a good idea to change the namespace to `bsp.jsonrpc` though, as, if this project could probably claim the `bsp` namespace in a global index, it'd be much less the case for `jsonrpc`. 

If/when something else defines its own jsonrpc semantics, an approach that I've used that kinda works is to pre-process the models to turn one set of semantics to another (when possible), to gain access to whatever that "something else" does.